machine-learning/incremental.ipynb

```diff
@@ -0,0 +1,542 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Incremental training of large datasets"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "A common approach to training large datasets is to incrementally train a model on many batches of data sequentially.  The Scikit-Learn documentation discusses this approach in more depth in their user guide, \"[Strategies to scale computationally: bigger data](http://scikit-learn.org/stable/modules/scaling_strategies.html)\".\n",
+    "\n",
+    "Many Scikit-Learn estimators implement a `partial_fit` method to enable incremental learning in batches.  Typically it is up to the user to collect batches of data to give to a Scikit-Learn estimator, which then updates a single estimator.\n",
+    "\n",
+    "```python\n",
+    "est = SGDClassifier()\n",
+    "\n",
+    "for filename in filenames:\n",
+    "    df = pandas.read_csv(filename)\n",
+    "    x, y = ...  # get data from CSV using the Pandas API\n",
+    "    est.partial_fit(x, y)\n",
+    "```\n",
```

Comment:
Do you have any examples for "this is a common operation"? The first thing I would do is shuffle the dataset before calling the optimizer on each block.

Reply:
> The biggest edits I made were adding some motivation for calling dask.persist.

Sounds fine to me.  As a general note, there are always going to be lots of things that we'll do in any of these examples that are important that we won't be able to tell them about.  We'll necessarily skip a lot of potential wisdom we could deliver in order to focus on the topic at hand.

In this case though I agree that including some information about persist is good, especially because it's nice and concise.